graph/api/src/main/java/org/jboss/windup/graph/model/ProjectModel.java

```diff
@@ -256,6 +266,33 @@ public ProjectModel getRootProjectModel()
         }
 
         @Override
+        public Iterable<ProjectModel> getApplications()
+        {
+            // The reframing is just to make sure we pass in the proxy and not the "$Impl" class instance
+            Vertex vertex = it();
+            ProjectModel reframed = frame(vertex, ProjectModel.class);
+            return this.getApplications(reframed);
+        }
+
+        private Set<ProjectModel> getApplications(ProjectModel project)
+        {
+            Set<ProjectModel> applications = new HashSet<>();
+            for (ProjectModel duplicate : project.getDuplicateProjects())
+            {
+                duplicate.getApplications().forEach(applications::add);
+            }
+
+            ProjectModel parent = project.getParentProject();
+            if (parent != null)
+                parent.getApplications().forEach(applications::add);
+
+            if (parent == null)
+                applications.add(project);
+
+            return applications;
+        }
```

Comment:
Naming is really confusing :( 

Do I understand it correctly, that in case of this structure:
```
A
|--- B
|--- C
```
getApplications() called on A would return A,  called on B would return B and A?

Reply:
I'm just adding a few notes from the discussion from this morning for reference.

Both would return "A" as the input application in this case is "A".